crates/rust-analyzer/src/handlers/request.rs

```diff
@@ -1127,30 +1128,34 @@ pub(crate) fn handle_completion_resolve(
     forced_resolve_completions_config.fields_to_resolve = CompletionFieldsToResolve::empty();
 
     let position = FilePosition { file_id, offset };
-    let Some(resolved_completions) = snap.analysis.completions(
+    let Some(completions) = snap.analysis.completions(
         &forced_resolve_completions_config,
         position,
         resolve_data.trigger_character,
     )?
     else {
         return Ok(original_completion);
     };
+
+    let Some(corresponding_completion) = completions.into_iter().find(|completion_item| {
+        let hash = completion_item_hash(completion_item, resolve_data.for_ref);
+        hash == resolve_data.hash
```

Comment:
I think may be a good idea to first check if the labels are identical and only compute and compare the hash if they are:

* this will reduce computational requirements a bit since hash is only computed for any realistic candidates
* as a bonus it makes collisons even more unlikely

Reply:
Nice idea.

It's a bit tricky because LSP completion items get their labels changed compared to the original ones, but luckily only by adding a suffix, so we can still reduce the computations with a `starts_with` check.